conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DefaultStaticFactoryMethodGenerator.java

```diff
@@ -65,30 +66,33 @@
 import java.util.Optional;
 import javax.lang.model.element.Modifier;
 
-public final class AsyncGenerator implements StaticFactoryMethodGenerator {
+public final class DefaultStaticFactoryMethodGenerator implements StaticFactoryMethodGenerator {
     private static final String REQUEST = "_request";
     private static final String PLAIN_SER_DE = "_plainSerDe";
 
     private final Options options;
     private final TypeNameResolver typeNameResolver;
     private final ParameterTypeMapper parameterTypes;
     private final ReturnTypeMapper returnTypes;
+    private final StaticFactoryMethodType methodType;
 
-    public AsyncGenerator(
+    public DefaultStaticFactoryMethodGenerator(
             Options options,
             TypeNameResolver typeNameResolver,
             ParameterTypeMapper parameterTypes,
-            ReturnTypeMapper returnTypes) {
+            ReturnTypeMapper returnTypes,
+            StaticFactoryMethodType methodType) {
         this.options = options;
         this.typeNameResolver = typeNameResolver;
         this.parameterTypes = parameterTypes;
         this.returnTypes = returnTypes;
+        this.methodType = methodType;
     }
 
     @Override
     public MethodSpec generate(ServiceDefinition def) {
```

Comment:
I kinda feel like there's a small refactoring to do with this class and everything, to stop having to pass params into methods, as that makes the code more bulky.

Just make everything into instance variables and simplify. But I don't really feel like doing this now.

Reply:
Or rather I don't mind doing this, just not in this PR.